packages/core/js-peer/src/index.ts

```diff
@@ -0,0 +1,222 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import type { ClientConfig, IFluenceClient, RelayOptions, ConnectionState, CallAquaFunctionType, RegisterServiceType } from '@fluencelabs/interfaces';
+import { ClientPeer, makeClientPeerConfig } from './clientPeer/ClientPeer.js';
+import { callAquaFunction } from './compilerSupport/callFunction.js';
+import { registerService } from './compilerSupport/registerService.js';
+import { MarineBackgroundRunner } from './marine/worker/index.js';
+// @ts-ignore
+import { BlobWorker } from 'threads';
+import * as fs from 'fs';
+import process from 'process';
+import { Buffer } from 'buffer';
+import { doRegisterNodeUtils } from './services/NodeUtils.js';
+import { createRequire } from 'module';
+
+const WORKER_VERSION = '__WORKER_VERSION__';
+const MARINE_VERSION = '__MARINE_VERSION__';
+const AVM_VERSION = '__AVM_VERSION__';
+// Override if necessary
+const CDN_ROOT = 'https://unpkg.com/';
+
+const isNode = typeof process !== 'undefined' && process?.release?.name === 'node';
+
+async function fetchResource(packageName: string, assetPath: string, version: string) {
+    if (isNode) {
+        const require = createRequire(import.meta.url);
+        
+        const file = await new Promise<Buffer>((resolve, reject) => {
+            // Cannot use 'fs/promises' with current vite config. This module is not polyfilled by default.
+            const workerFilePath = require.resolve(packageName + assetPath);
+            fs.readFile(workerFilePath, (err, data) => {
+                if (err) {
+                    reject(err);
+                    return;
+                }
+                resolve(data);
+            });
+        });
+        return new Response(file, {
+            headers: {
+                'Content-type':
+                    assetPath.endsWith('.wasm')
+                        ? 'application/wasm'
+                        : assetPath.endsWith('.js')
+                            ? 'application/javascript'
+                            : 'application/text'
+            }
+        });
+    } else {
+        // Will work everywhere fetch is supported.
+        return fetch(new globalThis.URL(`${packageName}@${version}${assetPath}`, CDN_ROOT));
```

Comment:
I think just using string instead of URL object should work. If not - please leave a comment why you have a URL object here

Reply:
Same reason we use path.join instead of combining path directly - less restriction on url segments.

Reply:
Also i thought about unified way to fetch asset - for example on node we can use file url. File url or http url could be wrapped in the same URL object and them we can use it on browser or node with the same code. Some kind of sync point for code. Alas, file url is not supported in node-fetch(yet).